x-pack/plugin/esql-core/src/main/java/org/elasticsearch/xpack/esql/core/type/KeywordEsField.java

```diff
@@ -53,6 +63,33 @@ protected KeywordEsField(
         this.normalized = normalized;
     }
 
+    private KeywordEsField(StreamInput in) throws IOException {
+        this(
```

Comment:
Not necessarily belonging to this PR, but the c'tor used here can be dropped and the one that doesn't require a DataType parameter used instead (was wondering why do we need to provide a type here).

Reply:
:+1: I'll save that one I think.